phoenix-core/src/main/java/org/apache/phoenix/schema/types/PBlob.java

```diff
@@ -17,15 +17,44 @@
  */
 package org.apache.phoenix.schema.types;
 
+import org.apache.phoenix.expression.BlobExpression;
+import org.apache.phoenix.schema.SortOrder;
+
+import java.io.IOException;
 import java.io.InputStream;
 import java.sql.Types;
 
+/**
+ * Phoenix representation of the BLOB datatype
+ */
 public class PBlob extends PVarbinary {
 
     public static final PBlob INSTANCE = new PBlob();
 
     private PBlob() {
-        super("BLOB", Types.BLOB, InputStream.class, null, 70);
+        super("BLOB", Types.BLOB, InputStream.class, null, 48);
+    }
+
+    @Override
+    public Object toObject(byte[] bytes, int offset, int length, PDataType actualType,
+            SortOrder sortOrder, Integer maxLength, Integer scale) {
+        Object b = super.toObject(bytes, offset, length, actualType, sortOrder, maxLength, scale);
+        if (!(b instanceof byte[])) {
+            throw new RuntimeException("Corrupted BLOB metadata found");
+        }
+        byte[] bytesCopy = (byte[])b;
+        BlobExpression.BlobMetaData blobMetaData;
+        try {
+            blobMetaData = BlobExpression.BlobMetaData.deserializeBlobMetaData(bytesCopy);
+        } catch (IOException | ClassNotFoundException e) {
+            throw new RuntimeException("Could not deserialize BLOB metadata");
+        }
+        if (blobMetaData != null) {
+            BlobExpression dummyBlobExpr = new BlobExpression(null, null, null);
+            // TODO: Should return a PhoenixBlob object rather than an InputStream
+            return dummyBlobExpr.getInputStream(blobMetaData);
```

Comment:
Same here. Will use PhoenixBlob after https://github.com/dbwong/phoenix/pull/5 is merged